pkg/utils/cluster.go

```diff
@@ -161,3 +171,94 @@ func AnnotateNode(ctx context.Context, nodeName string, key, value string, c cli
 
 	return AnnotateObject(ctx, node, key, value, c)
 }
+
+func FindNodePoolConfig(ctx context.Context, node *corev1.Node, c client.Client) (*sriovnetworkv1.SriovNetworkPoolConfig, []corev1.Node, error) {
```

Comment:
nit: add docstring

Reply:
also im thinking we should have two functions:

1. find node pool for node
2. find nodes for node pool (with special handling for case where default node pool was provided)

WDYT ?


Also please add UT for whatever we end up with

Reply:
+1